.github/workflows/checks.yml

```diff
@@ -97,53 +95,32 @@ jobs:
           poetry install
 
       # ====================== CONTRACTS v2 ====================== #
-
       - name: Cache contracts v2
-        id: cache-contracts-v2
+        id: cache-contracts_v2
         uses: actions/cache@v3
         with:
-          path: starknet_py/tests/e2e/mock/contracts_compiled_v2
-          key: ${{ runner.os }}-contracts-${{ hashFiles('starknet_py/tests/e2e/mock/contracts_v2') }}-${{ env.CAIRO_V2_SHA }}
-    
-      - uses: actions/checkout@v4
-        with:
-          repository: starkware-libs/cairo
-          path: cairo_v2
-          ref: ${{ env.CAIRO_V2_SHA }}
+          path: starknet_py/tests/e2e/mock/contracts_v2/target
```

Comment:
are you sure only `target` directory must be cached? what about the lock?

Reply:
Because of adding external contracts, I have changed my mind and the best solution will be to add scarb.lock to the repo to ensure that we use the same contracts locally and on CI. Then caching only `target` should be enough. WDYT?

Reply:
hmm, may make sense actually. I can see you're using file hashes for cac he keys, so this should work. :+1: